funnel/views/proposal.py

```diff
@@ -247,33 +247,44 @@ def edit(self) -> ReturnRenderWith:
     @requires_roles({'editor'})
     def add_collaborator(self) -> ReturnView:
         collaborator_form = ProposalMemberForm(proposal=self.obj)
-        if collaborator_form.validate_on_submit():
-            with db.session.no_autoflush:
-                membership = ProposalMembership(
-                    proposal=self.obj, granted_by=current_auth.user
-                )
-                collaborator_form.populate_obj(membership)
-                db.session.add(membership)
-            db.session.commit()
-            return {
-                'status': 'ok',
-                'message': _("{user} has been added as an collaborator").format(
-                    user=membership.user.pickername
-                ),
-                'html': render_template(
-                    'collaborator_list.html.jinja2',
-                    collaborators=[
-                        _m.current_access(datasets=['primary', 'related'])
-                        for _m in self.obj.memberships
-                    ],
-                ),
-            }, 201
+        if request.method == 'POST':
+            if collaborator_form.validate_on_submit():
```

Comment:
`form.validate_on_submit()` is shorthand for `request.method == 'POST' and form.validate()` so the first `if` statement is not required.